src/com/maddyhome/idea/vim/group/MotionGroup.java

```diff
@@ -1987,6 +1987,45 @@ public char getLastFTChar() {
     return lastFTChar;
   }
 
+  public int selectNextSearch(Editor editor, int count, boolean forwards) {
+    String lastSearch = VimPlugin.getSearch().getLastSearch();
+    int currentPos = editor.getCaretModel().getOffset();
+    TextRange nextRange = new TextRange(currentPos, currentPos);
+    for (int i = 0; i < count; i++) {
+      nextRange = VimPlugin.getSearch().findNext(editor, lastSearch, nextRange.getStartOffset(), false,forwards);
+      if (nextRange == null) {
+        return -1;
+      }
+    }
+
+    final int startOffset= startOffset(nextRange, forwards);
+    final int endOffset = endOffset(nextRange, forwards);
+    for (Caret caret : editor.getCaretModel().getAllCarets()) {
+      MotionGroup.moveCaret(editor, caret, startOffset);
+      setVisualMode(editor, CommandState.SubMode.VISUAL_CHARACTER);
+      MotionGroup.moveCaret(editor, caret, endOffset);
+    }
+    return endOffset;
+  }
+
+  private int startOffset(TextRange nextRange, boolean forwards) {
```

Comment:
Hmm, looks like `startOffset` and `endOffset` functions are used only in `selectNextSearch`. Maybe it would be better to use ternary operator instead without creating new functions?

Reply:
Ternary operator added, but methods added for readability. Specially after adding the check of the comment below.